scipy/stats/_continuous_distns.py

```diff
@@ -5831,6 +5832,21 @@ def _stats(self, c):
         excess_kurtosis = sc.polygamma(3, c) / (var*var)
         return mean, var, skewness, excess_kurtosis
 
+    def _entropy(self, c):
+        def regular(c):
+            h = sc.gammaln(c) - c * sc.digamma(c) + c
+            return h
+
+        def asymptotic(c):
+            # using asymptotic expansions for gammaln and psi (see gh-18093)
+            term_1 = -0.5 * np.log(c) + (1 / 6) * c ** -1.
+            term_2 = -(1 / 90) * c ** -3. + (1 / 210) * c ** -5.
+            h = norm._entropy() + term_1 + term_2
```

Comment:
The usual script agrees.
<details>

```python3
import sympy
from sympy import log, pi, Rational, simplify, expand
from IPython.display import display
x, c = sympy.symbols('x, c')

def digamma(x):
    return log(x) - 1/(2*x) - 1/(12*x**2) + 1/(120*x**4) - 1/(252*x**6)

def gammaln(x):
    return x*log(x) - x - log(x)/2 + log(2*pi)/2 + 1/(12*x) - 1/(360*x**3) + 1/(1260*x**5)

old = gammaln(c) - c * digamma(c) + c

display(expand(simplify(old)))
```

</details>

![image](https://github.com/scipy/scipy/assets/6570539/3c0a598d-cde7-4316-861c-3b317fcc81a3)